File: 백준/Bronze/18245. 이상한 나라의 암호/이상한 나라의 암호.py
import sys
input = lambda: sys.stdin.readline().rstrip()


def main():
    t = 1
    while 1:
        S = input()
        if S == "Was it a cat I saw?":
            break
        for i in range(0, len(S), t+1):
            print(S[i], end='')
        print()
        t += 1


if __name__ == '__main__':
    main()

File: 백준/Bronze/18245. 이상한 나라의 암호/이상한 나라의 암호.c
#include <stdio.h>
#include <string.h>

int main() {
    char S[10001];
    int t = 1;
    while (1) {
        scanf("%[^\n]%*c", S); 
        if (strcmp(S, "Was it a cat I saw?") == 0) {
            break;
        }
        for (int i = 0; i < strlen(S); i += t + 1) {
            putchar(S[i]);
        }
        putchar('\n');
        t++;
    }
    return 0;
}